operator/cmd/mesh/install.go

```diff
@@ -178,6 +180,11 @@ func Install(kubeClient kube.CLIClient, rootArgs *RootArgs, iArgs *InstallArgs,
 		return fmt.Errorf("failed to get profile, namespace or enabled components: %v", err)
 	}
 
+	// detect ambiet distroless tag
+	if profile == ambientProfile && strings.HasSuffix(tag, "-distroless") {
```

Comment:
Profile is just the higher level entrypoint to setting `variant`. Can we actually check `variant` itself?

You may need to do it in helm instead of go

Reply:
ACK

Reply:
I tried adding expressions like this in helm, and then found they might cause users to overlook their own mistakes.
`{{- if and (contains "-distroless" .Values.global.tag) (eq .Values.pilot.variant "distroless") }}`

then I feels it's more straightforward to throw validation errors during the generation config phase.